docs/docs/development/studio-setup.md

```diff
@@ -0,0 +1,134 @@
+| Y          | uint     | Key Y position (top-left point)      | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Rotation   | int      | Key rotation (positive => clockwise) | [centi-](https://en.wikipedia.org/wiki/Centi-)degree    |
+| Rotation X | int      | Rotation origin X position           | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Rotation Y | int      | Rotation origin Y position           | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+
+:::note
+You can specify negative values in devicetree using parentheses around it, e.g. `(-3000)` for a 30 degree counterclockwise rotation.
+:::
+
+## Header Include
+
+To pull in the necessary definition for creating physical layouts, a new include should be added to the top of the devicetree file:
+
+```
+#include <physical_layouts.dtsi>
+```
+
+## Example
+
+Here is an example physical layout for a 2x2 macropad:
+
+```dts
+    macropad_physical_layout: macropad_physical_layout {
+        compatible = "zmk,physical-layout";
+        display-name = "Macro Pad";
+
+        keys  //                     w   h    x    y     rot    rx    ry
+            = <&key_physical_attrs 100 100    0    0       0     0     0>
+            , <&key_physical_attrs 100 100  100    0       0     0     0>
+            , <&key_physical_attrs 100 100    0  100       0     0     0>
+            , <&key_physical_attrs 100 100  100  100       0     0     0>
+            ;
+    };
+```
+
+# Position Map
+
+When switching between layouts with ZMK Studio, the keymap of the previously selected layout is used to populate the keymap in the new layout. To determine which keymap entry maps to which entry in the new layout, keys between the two layouts that share the exact same physical attributes are matched.
+
+However, keys between layouts might not be in exactly the same positions, in which case a position map can be used. The position map includes a sequence for every relevant layout, and the corresponding entries in `positions` property will be used to determine the mapping between layouts. By default, the physical attribute matching behavior will be used as a fallback for positions not specified in the map, but the `complete` property can be added to the map to specify that no matching fallback should occur.
+
+## Examples
+
+### Basic Map
+
+For example, the following position map correctly maps the 5-column and 6-column Corne keymap layouts.
+
+```dts
+    foostan_corne_position_map {
+        compatible = "zmk,physical-layout-position-map";
+
+        complete;
+
+        twelve {
+            physical-layout = <&foostan_corne_6col_layout>;
+            positions
+                = < 1  2  3  4  5  6  7  8  9 10>
+                , <13 14 15 16 17 18 19 20 21 22>
+                , <25 26 27 28 29 30 31 32 33 34>
+                , <      36 37 38 39 40 41      >;
```

Comment:
I think this is missing the first and last columns by accident; shouldn't it be the same as the value in the "full preserving map" example?